storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -104,45 +110,68 @@
     stateRequestCachedCounter = stateRequestCounter.labels("cached");
     stateRequestRegenerateCounter = stateRequestCounter.labels("regenerate");
     stateRequestMissCounter = stateRequestCounter.labels("miss");
+
     this.time = time;
     this.genesis_time = genesis_time;
     this.justified_checkpoint = justified_checkpoint;
     this.finalized_checkpoint = finalized_checkpoint;
     this.best_justified_checkpoint = best_justified_checkpoint;
-    this.blocks = new ConcurrentHashMap<>(blocks);
+    this.blocks =
+        ConcurrentLimitedMap.create(
+            pruningOptions.getBlockCacheSize(), LimitStrategy.DROP_LEAST_RECENTLY_ACCESSED);
     this.block_states =
-        ConcurrentLimitedMap.create(stateCacheSize, LimitStrategy.DROP_LEAST_RECENTLY_ACCESSED);
-    this.checkpoint_states = new ConcurrentHashMap<>(checkpoint_states);
+        LimitedMap.create(
+            pruningOptions.getStateCacheSize(), LimitStrategy.DROP_LEAST_RECENTLY_ACCESSED);
+    this.checkpoint_states = new HashMap<>(checkpoint_states);
+
+    this.blockProvider =
+        BlockProvider.combined(
+            fromMap(Map.of(finalizedBlockAndState.getRoot(), finalizedBlockAndState.getBlock())),
```

Comment:
Doesn't this mean we wind up holding the finalised block and state from startup in memory permanently?

Reply:
good catch!  adding a helper utility to provide the latest finalized block dynamically